CHANGELOG.md

```diff
@@ -11,45 +11,118 @@ We use *breaking* word for marking changes that are not backward compatible (rel
+This to have SRV resolution working on [Golang 1.11+ with KubeDNS below v1.14](https://github.com/golang/go/issues/27546)
 
-New metrics:
-* `thanos_bucket_store_queries_dropped_total` shows how many queries were dropped due to the samples limit;
-* `thanos_bucket_store_queries_concurrent_max` is a constant metric which shows how many Series() calls can concurrently be executed by Thanos Store;
-* `thanos_bucket_store_queries_in_flight` shows how many queries are currently "in flight" i.e. they are being executed;
-* `thanos_bucket_store_gate_duration_seconds` shows how many seconds it took for queries to pass through the gate in both cases - when that fails and when it does not.
+   New Querier and Ruler flag: `-- store.sd-dns-resolver` which allows to specify resolver to use. Either `golang` or `miekgdns`
+   
+- [#986](https://github.com/improbable-eng/thanos/pull/986) Allow to save some startup & sync time in store gateway as it is no longer is needed to compute index-cache from block index on its own. 
+  For compatibility store GW still can do it, but it first checks bucket if there is index-cached uploaded already. 
+  In the same time, compactor precomputes the index cache file on every compaction. 
 
-New tracing span:
-* `store_query_gate_ismyturn` shows how long it took for a query to pass (or not) through the gate.
+  New Compactor flag: `--index.generate-missing-cache-file` was added to allow quicker addition of index cache files, if enabled it precomputes missing files on compactor startup. Note that it will take time and it's only one-off step per bucket.
 
 :warning: **WARNING** :warning: #798 adds a new default limit to Thanos Store: `--store.grpc.series-max-concurrency`. Most likely you will want to make it the same as `--query.max-concurrent` on Thanos Query.
 
-- [#970](https://github.com/improbable-eng/thanos/pull/970) Added `PartialResponseStrategy` field for `RuleGroups` for `Ruler`.
-- [#1016](https://github.com/improbable-eng/thanos/pull/1016) Added option for another DNS resolver (miekg/dns client). 
-This to have SRV resolution working on [Golang 1.11+ with KubeDNS below v1.14](https://github.com/golang/go/issues/27546)
-- [#986](https://github.com/improbable-eng/thanos/pull/986) Store index cache files in object storage, reduces store start-up time by skipping the generating the index cache for all blocks and only do this for recently created uncompacted blocks. 
+- [#887](https://github.com/improbable-eng/thanos/pull/887) Compact: Added new `--block-sync-concurrency` flag, which allows you tconfigure number of goroutines to use when syncing block metadata from object storage.
+
+- [#928](https://github.com/improbable-eng/thanos/pull/928) Query: Added `--store.response-timeout` flag. If a Store doesn't send any data in this specified duration then a Store will be ignored and partial data will be returned if it's enabled. 0 disables timeout.
```

Comment:
As we know this flag is not working that great due to blocking SeriesSet. I would mention that maybe?

Reply:
It does work at max it would timeout `--store.response-timeout` * Number of Thanos Store's in the case of all Thanos Stores are slow

Reply:
Which is Improbable case IMO :D

Reply:
Thiis is an rc release so I guess it's fine for now. I'll add docs & metrics in next one